core/bindings.rs

```diff
@@ -452,17 +452,46 @@ fn send(
     Err(_) => &[],
   };
 
-  let zero_copy: Option<ZeroCopyBuf> =
-    v8::Local::<v8::ArrayBufferView>::try_from(args.get(2))
-      .map(ZeroCopyBuf::new)
-      .ok();
-
   let state_rc = CoreIsolate::state(scope.isolate());
   let mut state = state_rc.borrow_mut();
   assert!(!state.global_context.is_empty());
 
   // If response is empty then it's either async op or exception was thrown
-  let maybe_response = state.dispatch_op(scope, op_id, control, zero_copy);
+  let maybe_response = match std::cmp::max(args.length() - 2, 0) {
+    0 => state.dispatch_op(scope, op_id, control, &mut []),
+    1 => {
+      let zero_copy_buf =
+        match v8::Local::<v8::ArrayBufferView>::try_from(args.get(2)) {
+          Ok(view) => ZeroCopyBuf::new(view),
+          Err(err) => {
+            let s = format!("invalid argument {}", err);
+            let msg = v8::String::new(scope, &s).unwrap();
+            scope.isolate().throw_exception(msg.into());
+            return;
+          }
+        };
+      state.dispatch_op(scope, op_id, control, &mut [zero_copy_buf])
+    }
+    cap => {
+      // Collect all ArrayBufferView's
+      let mut zero_copy_bufs =
+        Vec::with_capacity(usize::try_from(cap).unwrap());
+      for i in 2..args.length() {
+        zero_copy_bufs.push(
+          match v8::Local::<v8::ArrayBufferView>::try_from(args.get(i)) {
+            Ok(view) => ZeroCopyBuf::new(view),
+            Err(err) => {
+              let s = format!("invalid argument {} at {}", err, i - 2);
+              let msg = v8::String::new(scope, &s).unwrap();
+              scope.isolate().throw_exception(msg.into());
+              return;
+            }
+          },
+        );
+      }
+      state.dispatch_op(scope, op_id, control, &mut zero_copy_bufs)
+    }
+  };
```

Comment:
Any advice on how to reasonably reduce duplication here?

Reply:
Not so obvious indeed. This is what I could think of:

```rust
  let mut buf_iter = (2..args.length()).map(|i| {
    v8::Local::<v8::ArrayBufferView>::try_from(args.get(i))
      .map(ZeroCopyBuf::new)
      .map_err(|err| {
        let msg = format!("Invalid argument at position {}: {}", i, err);
        v8::String::new(scope, &msg).unwrap()
      })
  });
  let mut buf_one: ZeroCopyBuf; // Initialized only when there is one buffer.
  let mut buf_vec: Vec<ZeroCopyBuf>; // Initialized only when there is more than one buffer.
  let buf_iter_result = match buf_iter.len() {
    0 => Ok(&mut [][..]),
    1 => match buf_iter.next().unwrap() {
      Ok(b) => {
        buf_one = b;
        Ok(std::slice::from_mut(&mut buf_one))
      }
      Err(e) => Err(e),
    },
    _ => match buf_iter.collect::<Result<Vec<_>, _>>() {
      Ok(v) => {
        buf_vec = v;
        Ok(&mut buf_vec[..])
      }
      Err(e) => Err(e),
    },
  };
  let maybe_response = match buf_iter_result {
    Ok(mut bufs) => state.dispatch_op(scope, op_id, control, &mut bufs),
    Err(msg) => {
      scope.isolate().throw_exception(msg.into());
      return;
    }
  };
```

Reply:
I do think that it is a bit more efficient than pushing to a vector in a loop. However I'd be surprised if multi-buffer ops turn out to be a performance critical path.